shinylive/_main.py

```diff
@@ -469,6 +471,153 @@ def defunct_help(cmd: str) -> str:
 """
 
 
+# #############################################################################
+# ## shinylive.io url
+# #############################################################################
+
+
+@main.group(
+    short_help="Create or decode a shinylive.io URL.",
+    help="Create or decode a shinylive.io URL.",
+    no_args_is_help=True,
+    cls=OrderedGroup,
+)
+def url() -> None:
+    pass
+
+
+@url.command(
+    short_help="Create a shinylive.io URL from local files.",
+    help="""
+Create a shinylive.io URL for a Shiny app from local files.
+
+APP is the path to the primary Shiny app file.
+
+FILES are additional supporting files or directories for the app.
+
+On macOS, you can copy the URL to the clipboard with:
+
+    shinylive url encode app.py | pbcopy
+""",
+)
+@click.option(
+    "-m",
+    "--mode",
+    type=click.Choice(["editor", "app"]),
+    required=True,
+    default="editor",
+    help="The shinylive mode: include the editor or show only the app.",
+)
+@click.option(
+    "-l",
+    "--language",
+    type=click.Choice(["python", "py", "R", "r"]),
+    required=False,
+    default=None,
+    help="The primary language used to run the app, by default inferred from the app file.",
+)
+@click.option(
+    "-v", "--view", is_flag=True, default=False, help="Open the link in a browser."
+)
+@click.option(
+    "--no-header", is_flag=True, default=False, help="Hide the Shinylive header."
+)
+@click.argument("app", type=str, nargs=1, required=True, default="-")
+@click.argument("files", type=str, nargs=-1, required=False)
+def encode(
+    app: str,
+    files: Optional[tuple[str, ...]] = None,
+    mode: str = "editor",
+    language: Optional[str] = None,
+    no_header: bool = False,
+    view: bool = False,
+) -> None:
+    if app == "-":
+        app_in = sys.stdin.read()
+    else:
+        app_in = app
+
+    url = encode_shinylive_url(
+        app=app_in,
+        files=files,
+        mode=mode,
+        language=language,
+        header=not no_header,
+    )
+
+    print(url)
+
+    if view:
+        import webbrowser
+
+        webbrowser.open(url)
+    return
+
+
+@url.command(
+    short_help="Decode a shinylive.io URL.",
+    help="""
+Decode a shinylive.io URL.
+
+URL is the shinylive editor or app URL. If not specified, the URL will be read from
+stdin, allowing you to read the URL from a file or the clipboard.
+
+On macOS, you can read the URL from the clipboard with:
+
+    pbpaste | shinylive url decode
+""",
+)
+@click.option(
+    "--out",
```

Comment:
This should be `--dir` for consistency

```suggestion
    "--dir",
```